lib/lynx/config.rb

```diff
@@ -28,7 +28,7 @@ def [](key)
     end
 
     def detect(commands)
-      commands.detect{ |c| system("which #{c}") }
+      commands.detect{ |c| system("which #{c} >/dev/null") || exit(1)}
```

Comment:
I don't think we want to fail immediately here - the idea is to detect which `mysql` or `mysqldump` command works by iterating through the list. I would also be happy to remove this loop and just check for `mysql` and `mysqldump` only.